server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -455,6 +552,28 @@ public Query nullValueQuery() {
             return termQuery(nullValue(), null);
         }
 
+        @Override
+        public Query analyzePhrase(String field, TokenStream stream, int slop) throws IOException {
+            if (indexPhrases && slop == 0) {
+                stream = new FixedShingleFilter(stream, 2);
+                field = field + FAST_PHRASE_SUFFIX;
+            }
+            PhraseQuery.Builder builder = new PhraseQuery.Builder();
+            builder.setSlop(slop);
+
+            TermToBytesRefAttribute termAtt = stream.getAttribute(TermToBytesRefAttribute.class);
+            PositionIncrementAttribute posIncrAtt = stream.getAttribute(PositionIncrementAttribute.class);
+            int position = -1;
+
+            stream.reset();
+            while (stream.incrementToken()) {
+                position += posIncrAtt.getPositionIncrement();
```

Comment:
There is an option in `query_string` to disallow position increment (`enable_position_increments`)  so I guess that we should apply it here too.